contracts/SpokePoolV3Periphery.sol

```diff
@@ -12,169 +12,52 @@ import { IERC20Auth } from "./external/interfaces/IERC20Auth.sol";
+    // so this contract should also be able to be deterministically deployed at the same address across all networks
+    // since the periphery address is the only constructor argument.
+    SpokePoolV3Periphery public SPOKE_POOL_PERIPHERY;
 
-    /****************************************
-     *                ERRORS                *
-     ****************************************/
-    error MinimumExpectedInputAmount();
-    error LeftoverSrcTokens();
-    error InvalidFunctionSelector();
+    error LeftoverInputTokens();
+    error InvalidPeriphery();
     error ContractInitialized();
-    error InvalidMsgValue();
-    error InvalidSpokePool();
-    error InvalidSwapToken();
 
     /**
-     * @notice Construct a new SwapAndBridgeBase contract.
-     * @param _allowedSelectors Function selectors that are allowed to be called on the exchange.
+     * @notice Construct a new Proxy contract.
+     * @dev Is empty and all of the state variables are initialized in the initialize function
+     * to allow for deployment at a deterministic address via create2, which requires that the bytecode
+     * across different networks is the same. Constructor parameters affect the bytecode so we can only
+     * add parameters here that are consistent across networks.
      */
-    constructor(bytes4[] memory _allowedSelectors) {
-        for (uint256 i = 0; i < _allowedSelectors.length; i++) {
-            allowedSelectors[_allowedSelectors[i]] = true;
-        }
-    }
+    constructor() {}
 
     /**
-     * @notice Initializes the SwapAndBridgeBase contract.
-     * @param _spokePool Address of the SpokePool contract that we'll submit deposits to.
-     * @param _wrappedNativeToken Address of the wrapped native token for the network this contract is deployed to.
-     * @param _exchange Address of the exchange where tokens will be swapped.
-     * @dev These values are initialized in a function and not in the constructor so that the creation code of this contract
-     * is the same across networks with different addresses for the wrapped native token, the exchange this contract uses to
-     * swap and bridge, and this network's corresponding spoke pool contract. This is to allow this contract to be deterministically
-     * deployed with CREATE2.
-     * @dev This function can be front-run by anybody, so it is critical to check that the `spokePool`, `wrappedNativeToken`, and `exchange`
-     * values used in the single call to this function were passed in correctly before enabling the usage of this contract.
+     * @notice Initialize the SpokePoolPeripheryProxy contract.
+     * @param _spokePoolPeriphery Address of the SpokePoolPeriphery contract that this proxy will call.
      */
-    function initialize(
-        V3SpokePoolInterface _spokePool,
-        WETH9Interface _wrappedNativeToken,
-        address _exchange
-    ) external {
+    function initialize(SpokePoolV3Periphery _spokePoolPeriphery) external nonReentrant {
```

Comment:
Why an initialize function? Do we expect this to be an upgradable contract?

Reply:
We want to deploy this contract at the same create2 address everywhere so I wanted to avoid passing any constructor params that would modify the bytecode. Do you think that's not needed?

Reply:
Oh, I see.

I do think that makes sense! To avoid frontrunning, should we include a deployer contract that has no input params (and thus would have a consistent address) and has an function that creates the contract and initializes it atomically? This might also be useful since there's two contracts that are linked that we want to deploy together, so this deployer can ensure that both get deployed and setup correctly.

Reply:
Good call: 6db7d87b23924385680d9326ae52a96db82995d8